### PR TITLE
[common] add optional configmaps as part of library-charts

### PR DESCRIPTION
**Pull request by Cian911**
**Description of the change**

This PR allows for the creation of `ConfigMaps` as optional parameters in the values.yml file.
 

**Benefits**

Users will be able to create a `ConfigMap` for a project/chart, avoiding any manual steps to create one as described in an [example here](https://github.com/k8s-at-home/library-charts/pull/101#issuecomment-950162531).

**Possible drawbacks**

Currently this one provides a 1:1 mapping for a single `ConfigMap`.

**Applicable issues**

- Related issue: https://github.com/k8s-at-home/library-charts/issues/67

**Additional information**

N/A

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).

### CODE REVIEW

**Comment by bjw-s** [comment]
Hi! I've been away for a few days, but I'll try to take a look at the PR tomorrow 🙂

**Comment by Cian911** [comment]
> Hi! I've been away for a few days, but I'll try to take a look at the PR tomorrow 🙂

No problem @bjw-s :)

**Comment by bjw-s** [comment]
Sorry it took me so long to take a look at this! Thanks again for taking the time to create this PR! I think it looks great as a starting point. 👍 Would you mind if I add some commits to this PR to bring it more in line with the structure that is posited in #67?

File: charts/stable/common/templates/_configmap.tpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{{/*
+The ConfigMap object to be created.
```
Comment (bjw-s) [inline charts/stable/common/templates/_configmap.tpl:2]:
```suggestion
The ConfigMap object to be generated.
```

**Review by bjw-s — APPROVED** [review]

**Comment by bjw-s** [comment]
@allcontributors please add @Cian911 for code

**Comment by Cian911** [comment]
@bjw-s This is fantastic, no worries at all, and thanks for taking the time to review and change!

**Pull request merged**